src/ui/server/controllers/insurance/business/company-details/helpers/populate-companies-house-summary-list.ts

```diff
@@ -0,0 +1,44 @@
+import { companyHouseSummaryList } from '../../../../../helpers/summary-lists/company-house-summary-list';
+import mapSicCodes from '../../../../../helpers/mappings/map-sic-codes';
+import { FIELD_IDS } from '../../../../../constants';
+import { CompanyHouseResponse } from '../../../../../../types';
+
+const {
+  EXPORTER_BUSINESS: {
+    COMPANY_HOUSE: { COMPANY_NAME, COMPANY_NUMBER, COMPANY_INCORPORATED, COMPANY_SIC, COMPANY_ADDRESS },
+  },
+} = FIELD_IDS.INSURANCE;
+
+/**
+ * maps database response and returns body in correct format for summary list
+ * @param {object} exporterCompany
+ * @returns {CompanyHouseResponse} reformatted exporterCompany
+ */
+const mapDatabaseResponse = (exporterCompany: object) => {
+  return {
+    [COMPANY_NAME]: exporterCompany[COMPANY_NAME],
+    [COMPANY_ADDRESS]: {
+      ...exporterCompany[COMPANY_ADDRESS],
+    },
+    [COMPANY_NUMBER]: exporterCompany[COMPANY_NUMBER],
+    [COMPANY_INCORPORATED]: exporterCompany[COMPANY_INCORPORATED],
+    [COMPANY_SIC]: mapSicCodes(exporterCompany[COMPANY_SIC]),
+  } as CompanyHouseResponse;
+};
+
+/**
+ * receives exporterCompany and maps it to correct format and returns summaryList for company details
+ * @param {object} exporterCompany
+ * @returns {Array<SummaryListItemData>} summaryList for company details
+ */
+const populateCompaniesHouseSummaryList = (exporterCompany: object) => {
+  if (!exporterCompany[COMPANY_NUMBER]) {
+    return null;
+  }
+
+  const companyDetails = mapDatabaseResponse(exporterCompany);
+
+  return companyHouseSummaryList(companyDetails);
+};
+
```

Comment:
Could you check out this `helpers/summary-lists/answers-summary-list`, i'm working on the summary lists and trying to align everything.

This answers summary list is the same requirement we have for the other summary lists we'll be building soon; Answers summary list has very similar to this - a function that maps the data to content and is in `helpers/data-content-mappings`.

Currently in the "check answers" GET controller, summary list is generated like this:

```js
const answers = mapAnswersToContent(req.session.submittedData.quoteEligibility);

const summaryList = answersSummaryList(answers);
```


What do you think? Should we continue with the way answers are mapped and passed into summary list? Or we could do this as part of the `answersSummaryList` function

Either way I think the answers mapping to content should be in `helpers/data-content-mappings` and should name it similar to `mapAnswersToContent`; There is another one called `mapQuoteToContent`

Reply:
I think the check answers approach is correct, but I think this should have a separate ticket to get working